src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2187,21 +2188,32 @@ def aks_command_result(cmd, client, resource_group_name, name, command_id=""):
 
     commandResult = client.get_command_result(
         resource_group_name, name, command_id)
-    return commandResult
+    return _print_command_result(cmd.cli_ctx, commandResult)
 
 
-def _print_command_result(commandResult):
-    # succeed, print exitcode, and logs
-    if commandResult.provisioning_state == "Succeeded":
-        print(f"{colorama.Fore.GREEN}command started at {commandResult.started_at}, finished at {commandResult.finished_at}, with exitcode={commandResult.exit_code}{colorama.Style.RESET_ALL}")
-        print(commandResult.logs)
-        return
-    # failed, print reason in error
-    if commandResult.provisioning_state == "Failed":
-        print(f"{colorama.Fore.RED}command failed with reason: {commandResult.reason}{colorama.Style.RESET_ALL}")
-        return
-    # *-ing state
-    print(f"{colorama.Fore.BLUE}command is in : {commandResult.provisioning_state} state{colorama.Style.RESET_ALL}")
+def _print_command_result(cli_ctx, commandResult):
+    # cli_ctx.data['safe_params'] contains list of parameter name user typed in, without value.
+    # cli core also use this calculate ParameterSetName header for all http request from cli.
+    if cli_ctx.data['safe_params'] is None or "-o" in cli_ctx.data['safe_params'] or "--output" in cli_ctx.data['safe_params']:
+        # user specified output format, honor their choice, return object to render pipeline
+        return commandResult
+    else:
+        # user didn't specified any format, we can customize the print for best experience
+        if commandResult.provisioning_state == "Succeeded":
+            # succeed, print exitcode, and logs
+            print(f"{colorama.Fore.GREEN}command started at {commandResult.started_at}, finished at {commandResult.finished_at}, with exitcode={commandResult.exit_code}{colorama.Style.RESET_ALL}")
+            print(commandResult.logs)
```

Comment:
Please consider using styled text which could change based on color theme. Example is here: https://github.com/azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/util/custom.py#L178

Reply:
thanks, but AKS extension is already heavily using colorama.

having 2 different color/style would be more confusing.
switch to print_styled_text would be a big refactor to whole aks-preview. 

but will try it out internally and discuss with team if it's worth to switch.